### PR TITLE
Update docs with instructions on how to install examples from GitHub releases.

### PR DESCRIPTION
**Pull request by lbianchi-lbl**
As per #775 I've creates a dedicated page in the "FOQUS Installation and Running" section, between "Install FOQUS" and "Run FOQUS". It points to where the examples packages can be downloaded from GitHub, and that the archive should be extracted in a directory of the user's choice.

Since there are multiple places throughout the docs that refer to the examples as **examples/\<path-to-files-for-particular-example\>**, I've also added a suggestion for a template to use there. If you think something like this template could be useful, we can decide on a format, and then I'll go ahead and apply where needed.

Feedback is very welcome on both!

### CODE REVIEW

**Comment by lbianchi-lbl** [comment]
- Created substitution macro with reminder text pointing back to the "Install examples" section, which contains the steps to download and install the examples are explained
- Added reminder text inside a note box to all pages where the examples directory was referenced (as found by `grep examples/`)

For my side, work on this is complete and it should be ready to review (modulo the failing CircleCI tests that I'm looking into now).

**Comment by lbianchi-lbl** [comment]
Failing tests were caused by a dangling reference to the deleted file in the section's toctree (I wasn't aware about the useful `make dummy`). Good catch, CircleCI!

File: docs/source/chapt_install/install_examples.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.. _install_examples:
+
+Install FOQUS Examples
+----------------------
+
+.. note::
+   In previous releases the examples were packaged inside the main ``ccsi-foqus`` package.
+   Since version FIXME, they are no longer part of the ``ccsi-foqus`` package, but instead distributed as a separate archive.
```
Comment (ksbeattie) [inline docs/source/chapt_install/install_examples.rst:8]: It was version 3.5.0 that this was changed.

**Comment by lbianchi-lbl** [comment]
Build of the current docs with the relevant pages:

- "Install examples" page: https://lbl-lbianchi-foqus.readthedocs-hosted.com/en/feature-examples-docs/chapt_install/install_examples.html
- An example tutorial page, containing the reference pointing back to the "Install examples" page:  https://lbl-lbianchi-foqus.readthedocs-hosted.com/en/feature-examples-docs/chapt_HI/tutorial/HI_test.html#tutorial-heat-integration-with-foqus

**Review by ksbeattie — APPROVED** [review]
LGTM

**Pull request merged**